src/Components/SuperAdmins/Form/index.jsx

```diff
@@ -1,13 +1,84 @@
-import React from 'react';
-import { useState, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
+import { useParams, useHistory } from 'react-router-dom';
 import styles from './super-admins.module.css';
 
-const Form = (props) => {
-  if (!props.showForm) {
-    return null;
-  }
+const Form = () => {
+  const [superAdmins, setSuperAdmins] = useState([]);
+  const [isEdit, setIsEdit] = useState(false);
+  const { id } = useParams();
+  const history = useHistory();
 
-  const isPut = props.method === 'PUT';
+  const getSuperAdmins = async () => {
```

Comment:
You should use a getById, you save a lot of code like for example the one on line 90. You could remove the set from the useEffect and put it directly in the getById